src/index.js

```diff
@@ -273,7 +273,31 @@ function checkSgStatus(validSgs, statusResult, span, hasjson) {
         }
     }
     if (Object.keys(reasons).length) span?.setAttribute("details.sgsStatusCheck", JSON.stringify(reasons));
-    if (!hasjson && Object.keys(reasons).length === statusResult.length) throw "unhealthy subgraph";
+    if (!hasjson && Object.keys(reasons).length === statusResult.length) {
+        const urls = Object.keys(reasons);
+        const msg = ["subgraphs status check failed"];
+        if (urls.length === 1) {
+            // indexing error or invalid fulfilled response
+            if (typeof reasons[urls[0]] === "string") msg.push("Reason: " + reasons[urls[0]]);
+            else {
+                // AxsioError
+                if (reasons[urls[0]].message) msg.push("Reason: " + reasons[urls[0]].message);
```

Comment:
the problem with eliding {} on an if statement is that if you debug it later, you run the risk of things really not doing what you think they're doing

consider

```
if (reasons[urls[0]].message) 
console.log("foo")
msg.push("Reason: " + reasons[urls[0]].message);
```

the addition of the console log pushes the msg.push out of the if due to the meaningful whitespace

better to always wrap javascript code blocks in {}